components/contentselector/model/contentmodel.cpp

```diff
@@ -531,11 +544,95 @@ bool ContentSelectorModel::ContentModel::isEnabled (QModelIndex index) const
     return (flags(index) & Qt::ItemIsEnabled);
 }
 
-void ContentSelectorModel::ContentModel::setCheckStates (const QStringList &fileList, bool isChecked)
+bool ContentSelectorModel::ContentModel::isLoadOrderError(const EsmFile *file) const
+{
+    return mPluginsWithLoadOrderError.contains(file->filePath());
+}
+
+void ContentSelectorModel::ContentModel::setContentList(const QStringList &fileList, bool isChecked)
+{
+    mPluginsWithLoadOrderError.clear();
+    int previousPosition = -1;
+    foreach (const QString &filepath, fileList)
+    {
+        if (setCheckState(filepath, isChecked))
+        {
+            // as necessary, move plug-ins in visible list to match sequence of supplied filelist
+            const EsmFile* file = item(filepath);
+            int filePosition = indexFromItem(file).row();
+            if (filePosition < previousPosition)
+            {
+                mFiles.move(filePosition, previousPosition);
+                emit dataChanged(index(filePosition, 0, QModelIndex()), index(previousPosition, 0, QModelIndex()));
+            }
+            else
+            {
+                previousPosition = filePosition;
+            }
+        }
+    }
+    checkForLoadOrderErrors();
+}
+
+void ContentSelectorModel::ContentModel::checkForLoadOrderErrors()
 {
-    foreach (const QString &file, fileList)
+    for (int row = 0; row < mFiles.count(); ++row)
     {
-        setCheckState (file, isChecked);
+        EsmFile* file = item(row);
+        bool isRowInError = checkForLoadOrderErrors(file, row).count() != 0;
+        if (isRowInError)
+        {
+            mPluginsWithLoadOrderError.insert(file->filePath());
+        }
+        else
+        {
+            mPluginsWithLoadOrderError.remove(file->filePath());
+        }
+    }
+}
+
+QList<ContentSelectorModel::LoadOrderError> ContentSelectorModel::ContentModel::checkForLoadOrderErrors(const EsmFile *file, int row) const
+{
+    QList<LoadOrderError> errors = QList<LoadOrderError>();
+    foreach(QString dependentfileName, file->gameFiles())
+    {
+        const EsmFile* dependentFile = item(dependentfileName);
+
+        if (!dependentFile)
+        {
+            errors.append(LoadOrderError(LoadOrderError::ErrorCode_MissingDependency, dependentfileName));
+        }
+        if (!isChecked(dependentFile->filePath()))
+        {
+            errors.append(LoadOrderError(LoadOrderError::ErrorCode_InactiveDependency, dependentfileName));
+        }
+        if (row < indexFromItem(dependentFile).row())
+        {
+            errors.append(LoadOrderError(LoadOrderError::ErrorCode_LoadOrder, dependentfileName));
+        }
+    }
+    return errors;
+}
+
+QString ContentSelectorModel::ContentModel::toolTip(const EsmFile *file) const
+{
+    if (isLoadOrderError(file))
+    {
+        QString text("<font color=#840000><b>");
```

Comment:
Thanks for the feedback.  I've raised a bug
https://bugs.openmw.org/issues/2268 for it, and will see what I can do
about it.

On Sun, Jan 11, 2015 at 11:02 AM, scrawl <notifications@github.com> wrote:

> In components/contentselector/model/contentmodel.cpp
> <https://github.com/OpenMW/openmw/pull/421#discussion-diff-22762263>:
>
> > +        {
> > +            errors.append(LoadOrderError(LoadOrderError::ErrorCode_InactiveDependency, dependentfileName));
> > +        }
> > +        if (row < indexFromItem(dependentFile).row())
> > +        {
> > +            errors.append(LoadOrderError(LoadOrderError::ErrorCode_LoadOrder, dependentfileName));
> > +        }
> > +    }
> > +    return errors;
> > +}
> > +
> > +QString ContentSelectorModel::ContentModel::toolTip(const EsmFile *file) const
> > +{
> > +    if (isLoadOrderError(file))
> > +    {
> > +        QString text("<font color=#840000><b>");
>
> Another related problem: the red color for a content file in the table
> does *not* show up when that content file is selected. This could also be
> fixed by using an icon instead.
> http://scrawl.bplaced.net/perm/screenshot_23-01-07.png
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenMW/openmw/pull/421/files#r22762263>.
>